### PR TITLE
Google Analytics Features and Fixes

### PR DESCRIPTION
**Pull request by freiSMS**
## Proposed changes
**1) Feature**: Enable/ Disable automatic events via config
  - allows user to disable automatic events and prevent problems in custom reports:
    -https://community.jovo.tech/t/google-analytics-plugin-filter-events-in-custom-reports/1201?u=andre
  - fallback of "enableAutomaticEvents" is true to not affect developers using this feature 

**2) Fix**: incorrect page view tracking (state and intent) 
  - state was one state ahead
    - state always was set to the state of the response not request
    - in gAnalytics it looks like the user triggered (for example) "YesIntent" in "state2" - but he/she triggered it from "state1" and reached "state2" instead
    - -> mismatch of intent (from request) and state (from response)
  - intent tracking did track last called method not triggered intent

**3) Fix**: tracking to many session via timeouts
  - display devices are sending sessionEnded Request before starting a new session (even on next day)
  - gAnalytics did end such sessions (30 minutes or custom set timeout)
  - -> sessionEndedRequest are tracked as new session -> mismatch

**4) Feature**: Make usedID cryptographic
  - to match privacy guidelines the userId needs to be cryptographic
     - -> no connection from gAnalytics userId to alexaUserID
  - -> switched from murmurhash to sha256 hash

**5) Feature**: Add Alexa EndReason tracking
  - developers now can add "trackEndReasons = true" to track alexa endReasons
    - will extract endReason from alexaIntents and send it via custom metric
    - standard custom metrics are indizes 1 to 6 (can be configured) 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: jovo-integrations/jovo-analytics-googleanalytics/src/GoogleAnalytics.ts [hunk 1]
```diff
@@ -1,15 +1,61 @@
...
+   * Get end reason from session variables
+   *
+   * @param jovo - unser liebes Jovo objekt.
+   * @returns - the endreason saved in the session data.
+   */
+  static getEndReason(jovo: Jovo): validEndReasons | undefined {
+    const endReason: validEndReasons | undefined = jovo.$session.$data.endReason;
+    return endReason;
+  }
+  readonly sessionTimeoutInMinutes = 5;
```
Comment (aswetlow) [inline jovo-integrations/jovo-analytics-googleanalytics/src/GoogleAnalytics.ts:40]: Could you make this one configurable?

**Comment by aswetlow** [comment]
Thank you, André!

File: jovo-integrations/jovo-analytics-googleanalytics/src/GoogleAnalytics.ts (hunk 1 above)
Comment (freiSMS) [reply jovo-integrations/jovo-analytics-googleanalytics/src/GoogleAnalytics.ts]: done

**Comment by freiSMS** [comment]
added sessionTimeout in minutes to config

File: jovo-integrations/jovo-analytics-googleanalytics/package.json [hunk 2]
```diff
@@ -16,10 +16,12 @@
 	"license": "Apache-2.0",
 	"dependencies": {
 		"jovo-core": "^3.0.20",
+		"jovo-framework": "^3.0.22",
 		"lodash.get": "^4.4.2",
 		"lodash.merge": "^4.6.2",
 		"murmurhash": "0.0.2",
```
Comment (aswetlow) [inline jovo-integrations/jovo-analytics-googleanalytics/package.json:22]: Could you remove the `murmurhash` dependency? It's not used anymore, right?

**Comment by freiSMS** [comment]
I deleted murmurhash like you requested @aswetlow . I also changed the sending logic like omenocal suggested here https://community.jovo.tech/t/google-analytics-plugin-filter-events-in-custom-reports/1201?u=andre. Events are not sent anymore directly - but collected and sent together.

File: jovo-integrations/jovo-analytics-googleanalytics/package.json (hunk 2 above)
Comment (freiSMS) [reply jovo-integrations/jovo-analytics-googleanalytics/package.json]: done @aswetlow

**Comment by freiSMS** [comment]
@aswetlow @jankoenig I also added a googleAssistant feature to filter no voicematch users and google crawlers. To me these two were responsible for not tracking googleAssistant via googleAnalytics at all.
Both are corrupting the user and session count.

**Pull request merged**